store/store.go

```diff
@@ -193,6 +207,25 @@ func GetParams(q url.Values, schema *SchemaVersion) (*Params, error) {
 		SampleIntervalMinimum: sampleIntervalMinimum,
 	}
 
+	// Parse the allowed filters to further restrict the result set,
+	// e.g. "dosingDecision.reason=normalBolus,simpleBolus,watchBolus" to filter out dosing decisions
```

Comment:
Could we use an existing OpenAPI spec parameter format, maybe `style: deepObject` & `explode: true`? Would the codegen do the right thing in that case?

https://swagger.io/docs/specification/v3_0/serialization/#query-parameters

Reply:
This syntax mimics mongo's query syntax and the clinic service filters use that as well. I don't want to introduce a new "style" as part of this PR.